src/js/stores/MesosStateStore.js

```diff
@@ -256,7 +256,13 @@ class MesosStateStore extends GetSetBaseStore {
     }
 
     return tasks
-      .filter(task => task.isStartedByMarathon && task.name === mesosTaskName)
+      .filter(
+        task =>
+          task.isStartedByMarathon &&
+          task.name === mesosTaskName &&
```

Comment:
In our case tasks names do match. Name is `nginx` for both framework and the pod instance. Though Task IDs are different. `nginx-busybox.instance-92534c81-128a-11e9-aae4-ba5756c7ca71.nginx` I wonder if we could leverage this.

Reply:
https://github.com/dcos/dcos-ui/pull/3501/commits/2f8699e2cce84f5831b2aca83ffc362844e5fc23#diff-22eec13077e445e982cd1a9232667028R233
```javascript
const mesosTaskName = service.getMesosId();
```
Fun fact. We call `MesosId` and assign the result to `taskName`. But if you look inside `getMesosId` you'd be surprised that this is UI only hoodoo 🤷‍♂️

Reply:
I'd suggest we try telling _pods_ tasks from other tasks instead. 

The trouble is that if you have a pod with a container named like another task we have the issue. But you can' really have i.e. framework called `nginx` and single container app called `nginx`

Reply:
Yeah, I noticed the different task IDs, but I thought that the discovery.name was a cleaner solution, because I didn't know it was optional. I will try thinking of something else. Thank you for your help.

Reply:
I noticed that only pods tend to have `executor_id`, but it's marked as optional in `mesos.proto`, so I think that frameworks might have `executor_id` as well.

Reply:
The task id for the nginx framework looks like
```
task_id: Object { value: "nginx.b21e9bc9-1337-11e9-91f2-5e1fefcf97fb" }
```
The task id for a pod with a container image named "nginx" is:
```
task_id: Object { value: "nginx.instance-96331628-1337-11e9-91f2-5e1fefcf97fb.nginx" }
```
and the task id for a pod with a container image containing a dot is
```
task_id: Object { value: "nginx.instance-1.instance-1181ba75-1340-11e9-91f2-5e1fefcf97fb.nginx" }
```
So we can't just take the first part before the dot, because that might be misleading. I noticed that only the pods had a task_id ending with the container name, so I try to use that.